pandas/tests/io/test_sql.py

```diff
@@ -3036,426 +3308,337 @@ def dtype_backend_expected(self, storage, dtype_backend) -> DataFrame:
-            tm.assert_frame_equal(res1, res2)
-
-    def test_self_join_date_columns(self):
-        # GH 44421
-        from sqlalchemy.engine import Engine
-        from sqlalchemy.sql import text
-
-        create_table = text(
-            """
-        CREATE TABLE person
-        (
-            id serial constraint person_pkey primary key,
-            created_dt timestamp with time zone
-        );
-
-        INSERT INTO person
-            VALUES (1, '2021-01-01T00:00:00Z');
-        """
-        )
-        if isinstance(self.conn, Engine):
-            with self.conn.connect() as con:
-                with con.begin():
-                    con.execute(create_table)
-        else:
-            with self.conn.begin():
-                self.conn.execute(create_table)
-
-        sql_query = (
-            'SELECT * FROM "person" AS p1 INNER JOIN "person" AS p2 ON p1.id = p2.id;'
-        )
-        result = pd.read_sql(sql_query, self.conn)
-        expected = DataFrame(
-            [[1, Timestamp("2021", tz="UTC")] * 2], columns=["id", "created_dt"] * 2
+            if_exists="append",
         )
-        tm.assert_frame_equal(result, expected)
-
-        # Cleanup
-        with sql.SQLDatabase(self.conn, need_transaction=True) as pandasSQL:
-            pandasSQL.drop_table("person")
+        == 2
+    )
+    res = sql.read_sql_table("test_schema_other", conn, schema="other")
+    tm.assert_frame_equal(concat([df, df], ignore_index=True), res)
 
 
-# -----------------------------------------------------------------------------
-# -- Test Sqlite / MySQL fallback
 
+@pytest.mark.db
+def test_self_join_date_columns(postgresql_psycopg2_engine):
+    # GH 44421
+    conn = postgresql_psycopg2_engine
+    from sqlalchemy.engine import Engine
+    from sqlalchemy.sql import text
 
-class TestSQLiteFallback(SQLiteMixIn, PandasSQLTest):
-    """
-    Test the fallback mode against an in-memory sqlite database.
+    create_table = text(
+        """
+    CREATE TABLE person
+    (
+        id serial constraint person_pkey primary key,
+        created_dt timestamp with time zone
+    );
 
+    INSERT INTO person
+        VALUES (1, '2021-01-01T00:00:00Z');
     """
+    )
+    with conn.connect() as con:
+        with con.begin():
+            con.execute(create_table)
 
-    flavor = "sqlite"
-
-    @pytest.fixture(autouse=True)
-    def setup_method(self, iris_path, types_data):
-        self.conn = self.connect()
-        self.load_iris_data(iris_path)
-        self.load_types_data(types_data)
-        self.pandasSQL = sql.SQLiteDatabase(self.conn)
+    sql_query = (
+        'SELECT * FROM "person" AS p1 INNER JOIN "person" AS p2 ON p1.id = p2.id;'
+    )
+    result = pd.read_sql(sql_query, conn)
+    expected = DataFrame(
+        [[1, Timestamp("2021", tz="UTC")] * 2], columns=["id", "created_dt"] * 2
+    )
+    tm.assert_frame_equal(result, expected)
 
-    def test_read_sql_parameter(self, sql_strings):
```

Comment:
read_sql_parameter and read_sql_named_parameter already exist in `PandasTest`, so still have to parametrize / convert that test but will render these unnecessary